### PR TITLE
chore: log data for machine init and machine start

### LINKED ISSUE #4213: Podman machine telemetry
### Is your feature request related to a problem? Please describe

Today we are not tracking the success of creating or starting Podman machines. We should do this
- In order to get data that we can use to improve Podman Desktop,
- provide feedback to the Podman team, and
- learn a bit more about what will be useful to track in other telemetry and debug errors.

### Describe the solution you'd like

Send telemetry events for podman machine init & start. Each should send a single event _after_ the operation with information about the execution and success/failure. Specifically:

- Confirm basic info (OS, PD version) to be covered in every event.
- Include other environment info that may help understand failure cases or performance differences, e.g. WSL version.
- For failures, capture error code/message. We may need to strip/replace text in messages to avoid things like paths that would either be PII or cause every error to look different. (Is there anything else we can capture here to differentiate between success/failure/usability problem? Does it look actionable?)
- Include execution time, so that we can gauge performance.
- Double-check there is no PII in what we capture.
- Check the Podman team to see if there's anything else that would be useful for them to debug problems.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
log events to analyze the failures on `podman machine init` and `podman machine start`

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/4213

### How to test this PR?

unit tests have been added

### CODE REVIEW

**Review by deboer-tim — APPROVED** [review]

File: extensions/podman/src/podman-binary-location-helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  return {
+    process: {
+      exec: vi.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  podmanBinaryLocationHelper = new PodmanBinaryLocationHelper();
+  vi.resetAllMocks();
```
Comment (dgolovin) [inline extensions/podman/src/podman-binary-location-helper.spec.ts:38]: mockReset is a mockClear+ function, meaning if you reset, you don't need to clear after that.
https://vitest.dev/api/mock.html#mockreset

File: extensions/podman/src/wsl-helper.spec.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    process: {
+      exec: vi.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  wslHelper = new WslHelper();
+  vi.resetAllMocks();
+  vi.clearAllMocks();
```
Comment (dgolovin) [inline extensions/podman/src/wsl-helper.spec.ts]:
```suggestion
```

File: extensions/podman/src/qemu-helper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    process: {
+      exec: vi.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  qemuHelper = new TestQemuHelper();
+  vi.resetAllMocks();
+  vi.clearAllMocks();
```
Comment (dgolovin) [inline extensions/podman/src/qemu-helper.spec.ts]:
```suggestion
```

File: extensions/podman/src/podman-info-helper.spec.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    process: {
+      exec: vi.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  podmanInfoHelper = new PodmanInfoHelper();
+  vi.resetAllMocks();
+  vi.clearAllMocks();
```
Comment (dgolovin) [inline extensions/podman/src/podman-info-helper.spec.ts]:
```suggestion
```

File: extensions/podman/src/podman-binary-location-helper.spec.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  vi.clearAllMocks();
```
Comment (dgolovin) [inline extensions/podman/src/podman-binary-location-helper.spec.ts]:
```suggestion
```

**Review by dgolovin — APPROVED** [review]

File: extensions/podman/src/podman-binary-location-helper.spec.ts (hunk 1 above)
Comment (benoitf) [reply extensions/podman/src/podman-binary-location-helper.spec.ts:38]: I think we'll have to amend lot of spec.ts files as I did a copy/paste. Fixed

File: extensions/podman/src/wsl-helper.spec.ts (hunk 2 above)
Comment (benoitf) [reply extensions/podman/src/wsl-helper.spec.ts]: fixed

File: extensions/podman/src/qemu-helper.spec.ts (hunk 3 above)
Comment (benoitf) [reply extensions/podman/src/qemu-helper.spec.ts]: fixed

File: extensions/podman/src/podman-binary-location-helper.spec.ts (hunk 5 above)
Comment (benoitf) [reply extensions/podman/src/podman-binary-location-helper.spec.ts]: fixed

File: extensions/podman/src/podman-info-helper.spec.ts (hunk 4 above)
Comment (benoitf) [reply extensions/podman/src/podman-info-helper.spec.ts]: fixed

**Pull request merged**